src/AutoMapper.Collection/Mappers/EquivalentExpressionAddRemoveCollectionMapper.cs

```diff
@@ -62,30 +62,45 @@ public class EquivalentExpressionAddRemoveCollectionMapper : IConfigurationObjec
             return destination;
         }
 
-        private static readonly MethodInfo MapMethodInfo = typeof(EquivalentExpressionAddRemoveCollectionMapper).GetRuntimeMethods().First(_ => _.IsStatic);
+        private static readonly MethodInfo _mapMethodInfo = typeof(EquivalentExpressionAddRemoveCollectionMapper).GetRuntimeMethods().Single(x => x.IsStatic && x.Name == nameof(Map));
 
         public bool IsMatch(TypePair typePair)
         {
             return typePair.SourceType.IsEnumerableType()
-                   && typePair.DestinationType.IsCollectionType()
-                   && this.GetEquivalentExpression(TypeHelper.GetElementType(typePair.SourceType), TypeHelper.GetElementType(typePair.DestinationType)) != null;
+                   && typePair.DestinationType.IsCollectionType();
         }
 
         public Expression MapExpression(IConfigurationProvider configurationProvider, ProfileMap profileMap, IMemberMap memberMap,
             Expression sourceExpression, Expression destExpression, Expression contextExpression)
         {
+            IObjectMapper nextMapper = _collectionMapper;
+            var typePair = new TypePair(sourceExpression.Type, destExpression.Type);
+            var mappers = new List<IObjectMapper>(configurationProvider.GetMappers());
+            for (var i = mappers.IndexOf(this) + 1; i < mappers.Count; i++)
+            {
+                var mapper = mappers[i];
+                if (mapper.IsMatch(typePair))
+                {
+                    nextMapper = mapper;
+                    break;
+                }
+            }
+            var nextMapperExpression = nextMapper.MapExpression(configurationProvider, profileMap, memberMap, sourceExpression, destExpression, contextExpression);
```

Comment:
Can we put this in after the check failed.  That way if there's equivalency we aren't wasting time checking for the next mapper.